quantum/split_common/split_util.c

```diff
@@ -59,14 +59,15 @@ volatile bool isLeftHand = true;
 #if defined(SPLIT_USB_DETECT)
 _Static_assert((SPLIT_USB_TIMEOUT / SPLIT_USB_TIMEOUT_POLL) <= UINT16_MAX, "Please lower SPLIT_USB_TIMEOUT and/or increase SPLIT_USB_TIMEOUT_POLL.");
 static bool usbIsActive(void) {
+    bool is_active = false;
     for (uint16_t i = 0; i < (SPLIT_USB_TIMEOUT / SPLIT_USB_TIMEOUT_POLL); i++) {
         // This will return true if a USB connection has been established
         if (usb_connected_state()) {
-            return true;
+            is_active = true;
         }
         wait_ms(SPLIT_USB_TIMEOUT_POLL);
     }
-    return false;
+    return is_active;
 }
```

Comment:
This should be reverted -- we want it to exit early when connected. Your changes unconditionally continue to call the delay.

Reply:
> This should be reverted -- we want it to exit early when connected. Your changes unconditionally continue to call the delay.

Even if the master side exits early when USB connected, the slave is stuck in this delay loop. Behavior wise, it's the same, no split comms until the delay is over. However, this patch avoids the initial packet spam from master while the slave is still looking to see if it's USB is active ( no effective payload getting exchanged).

Reply:
> This should be reverted -- we want it to exit early when connected. Your changes unconditionally continue to call the delay.

I 2nd this. Now if someone is only using a single half of the keyboard they encounter a delay. One that would effect things like getting into bios.